README.md
# Invasion

/////////////////////
0.1.4a (hotfix)
addon settings were behaving strange, had to redo them again
/////////////////////
0.1.4
Added more enemy's to towns to include APC's and AA units so watch out when flying
changed roles to include Medic's Engineers and many more special people
reactivated Vcom, hopefully without ragdolling the AI will die. 
reworked ace addon settings
reworked the arsonel at base, (has more stuff now)
added humvee's to the base

/////////////////////
0.1.1
Nighttime only lasts a hour starting at 2100 time increases till 0500 then time slows back down.
Added more lights at base so its not so dark at night. (untactical but it helps see where your going)
added pictures to the whiteboard and loading screen.

/////////////////////
1.0
Fixed towns not being able to be capped,
Added dynamic weather,
Added time multiplier for faster day night
Made civilians spawn in all towns and villages

/////////////////////
0.0.7
hotfix for enemy units ragdolling and getting back up, (issue with ace and Vcom/ Fix for now is to get rid of Vcom.)

/////////////////////
0.0.6
have added sector control, just for tasking. 
reworked the base a littlebit, 
added VA to inside base, next to officer to recruit soldiers at.

/////////////////////
0.0.5 Initial Release

changelog.txt
player createDiaryRecord ["Diary",["Situation","

/////////////////////<br/>
0.1.4
Added more enemy's to towns to include APC's and AA units so watch out when flying
changed roles to include Medic's Engineers and many more special people
reactivated Vcom, hopefully without ragdolling the AI will die. 
reworked ace addon settings
reworked the arsonel at base, (has more stuff now)
added humvee's to the base

/////////////////////<br/>
0.1.1<br/>
Nighttime only lasts a hour starting at 2100 time increases till 0500 then time slows back down.<br/>
Added more lights at base so its not so dark at night. (untactical but it helps see where your going)<br/>
added pictures to the whiteboard and loading screen.<br/>
<br/>
/////////////////////<br/>
1.0<br/>
Fixed towns not being able to be capped,<br/>
Added dynamic weather,<br/>
Added time multiplier for faster day night<br/>
Made civilians spawn in all towns and villages<br/>
<br/>
/////////////////////<br/>
0.0.7<br/>
hotfix for enemy units ragdolling and getting back up, (issue with ace and Vcom/ Fix for now is to get rid of Vcom.)<br/>
<br/>
/////////////////////<br/>
0.0.6<br/>
have added sector control, just for tasking. <br/>
reworked the base a littlebit, <br/>
added VA to inside base, next to officer to recruit soldiers at.<br/>
<br/>
/////////////////////<br/>
0.0.5 Initial Release"]];



userconfig/VCOM_AI/AISettingsV4.hpp

/*
	ADDITIONAL COMMANDS
	(group this) setVariable ["VCM_NOFLANK",true]; //This command will stop the AI squad from executing advanced movement maneuvers.
	(group this) setVariable ["VCM_NORESCUE",true]; //This command will stop the AI squad from responding to calls for backup.
	(group this) setVariable ["VCM_TOUGHSQUAD",true]; //This command will stop the AI squad from calling for backup.
	(group this) setVariable ["Vcm_Disable",true]; //This command will disable Vcom AI on a group entirely.
	(group this) setVariable ["VCM_DisableForm",true]; //This command will disable AI group from changing formations.
	(group this) setVariable ["VCM_Skilldisable",true]; //This command will disable an AI group from being impacted by Vcom AI skill changes.

*/

Vcm_ActivateAI = true; //Set this to false to disable VcomAI. It can be set to true at any time to re-enable Vcom AI
VcmAI_ActiveList = []; //Leave this alone.
VCM_AIMagLimit = 2; //Number of mags remaining before AI looks for ammo.
VCM_Debug = false; //Enable debug mode.
VCM_MINECHANCE = 10; //Chance to lay a mine
VCM_LGARRISONCHANCE = 20; //Chance to perform a temporary garrison. 0 = disabled


//VCOM ARTILLERY. Only one kind of advanced artillery can be used at a time.
VCM_ARTYENABLE = false; //Enable improved artillery handling from Vcom.
VCM_ARTYSIDES = [west,east,resistance];  //Sides that will use VCOM/FFE artillery
VCM_ARTYLST = []; //List of all AI inside of artillery pieces, leave this alone.
VCM_ARTYDELAY = 300; //Delay between squads requesting artillery
VCM_ARTYWT = -(VCM_ARTYDELAY);
VCM_ARTYET = -(VCM_ARTYDELAY);
VCM_ARTYRT = -(VCM_ARTYDELAY);
VCM_ARTYSPREAD = 400; //Spread of artillery rounds;
Vcm_ArtilleryArray = []; //Leave this alone

//Fire For Effect Artillery handling. Only one kind of advanced artillery can be used at a time. - https://forums.bohemia.net/forums/topic/159152-fire-for-effect-the-god-of-war-smart-simple-ai-artillery/
VCM_FFEARTILLERY = true;

VCM_SIDEENABLED = [west,east,resistance]; //Sides that will activate Vcom AI
VCM_RAGDOLL = false; //Should AI ragdoll when hit
VCM_RAGDOLLCHC = 50; //CHANCE AI RAGDOLL
VCM_AIHEALING = true; //Makes AI heal themselves
VCM_FullSpeed = true; //Enforce full speedmode during combat (Does not reset after combat end)
VCM_HEARINGDISTANCE = 800; //Distance AI hear unsuppressed gunshots.
VCM_WARNDIST = 1000; //How far AI can request help from other groups.
VCM_WARNDELAY = 30; //How long the AI have to survive before they can call in for support. This activates once the AI enter combat.
VCM_STATICARMT = 300; //How long AI stay on static weapons when initially arming them. This is just for AI WITHOUT static bags. They will stay for this duration when NO ENEMIES ARE SEEN, or their group gets FAR away.
VCM_StealVeh = true; //Will the AI steal vehicles.
VCM_AIDISTANCEVEHPATH = 100; //Distance AI check from the squad leader to steal vehicles
VCM_ADVANCEDMOVEMENT = true; //True means AI will actively generate waypoints if no other waypoints are generated for the AI group (2 or more). False disables this advanced movements.
VCM_FRMCHANGE = true; //AI GROUPS WILL CHANGE FORMATIONS TO THEIR BEST GUESS.
VCM_SKILLCHANGE = true; //AI Groups will have their skills changed by Vcom.

//VCOM DRIVING. Experimental feature

Vcm_DrivingActivated = false; //Set this to false to disable VcomAI driving from executing.
VCM_DrivingDist = 15; // Distance from predicted path to search for objects. High numbers may cause instability.
VCM_DrivingDelay = 0.75; // How often the script should look for obstacles
VCM_DRIVERLIMIT = 4; // How many drivers should be calculated at each cycle
	
//AI SKILL SETTINGS HERE!!!!!!!!!!!!
//LOW DIFFICULTY
//VCM_AIDIFA = [['aimingAccuracy',0.15],['aimingShake',0.1],['aimingSpeed',0.25],['commanding',1],['courage',1],['endurance',1],['general',1],['reloadSpeed',1],['spotDistance',0.85],['spotTime',0.85]];

//MEDIUM DIFFICULTY
VCM_AIDIFA = [['aimingAccuracy',0.25],['aimingShake',0.15],['aimingSpeed',0.35],['commanding',1],['courage',1],['endurance',1],['general',1],['reloadSpeed',1],['spotDistance',0.85],['spotTime',0.85]];

//HIGH DIFFICULTY
//VCM_AIDIFA = [['aimingAccuracy',0.35],['aimingShake',0.4],['aimingSpeed',0.45],['commanding',1],['courage',1],['endurance',1],['general',1],['reloadSpeed',1],['spotDistance',0.85],['spotTime',0.85]];

//VCOM AI Skill Variables. Default values set to medium difficulty (old default)
//AIMING ACCURACY
VCM_AISKILL_AIMINGACCURACY_W = 0.25;
VCM_AISKILL_AIMINGACCURACY_E = 0.25;
VCM_AISKILL_AIMINGACCURACY_R = 0.25;

//AIMING SHAKE
VCM_AISKILL_AIMINGSHAKE_W = 0.15;
VCM_AISKILL_AIMINGSHAKE_E = 0.15;
VCM_AISKILL_AIMINGSHAKE_R = 0.15;

//AIMING SPEED
VCM_AISKILL_AIMINGSPEED_W = 0.35;
VCM_AISKILL_AIMINGSPEED_E = 0.35;
VCM_AISKILL_AIMINGSPEED_R = 0.35;

VCM_AISIDESPEC =
{
	//Reallocates skill variables before group skill settings are applied
	VCM_AIDIFWEST = [['aimingAccuracy',VCM_AISKILL_AIMINGACCURACY_W],['aimingShake',VCM_AISKILL_AIMINGSHAKE_W],['aimingSpeed',VCM_AISKILL_AIMINGSPEED_W],['commanding',1],['courage',1],['endurance',1],['general',1],['reloadSpeed',1],['spotDistance',0.85],['spotTime',0.85]];
	VCM_AIDIFEAST = [['aimingAccuracy',VCM_AISKILL_AIMINGACCURACY_E],['aimingShake',VCM_AISKILL_AIMINGSHAKE_E],['aimingSpeed',VCM_AISKILL_AIMINGSPEED_E],['commanding',1],['courage',1],['endurance',1],['general',1],['reloadSpeed',1],['spotDistance',0.85],['spotTime',0.85]];
	VCM_AIDIFRESISTANCE = [['aimingAccuracy',VCM_AISKILL_AIMINGACCURACY_R],['aimingShake',VCM_AISKILL_AIMINGSHAKE_R],['aimingSpeed',VCM_AISKILL_AIMINGSPEED_R],['commanding',1],['courage',1],['endurance',1],['general',1],['reloadSpeed',1],['spotDistance',0.85],['spotTime',0.85]];
	
	private _Side = (side (group _this));
	switch (_Side) do {
		case west:
		{
			{
				_this setSkill _x;
			} forEach VCM_AIDIFWEST;
		};
		case east:
		{
			{
				_this setSkill _x;
			} forEach VCM_AIDIFEAST;
		};
		case resistance:
		{
			{
				_this setSkill _x;
			} forEach VCM_AIDIFRESISTANCE;
		};
	};
};


VCM_CLASSNAMESPECIFIC = false; //Do you want the AI to have classname specific skill settings?
VCM_SIDESPECIFICSKILL = true; //Do you want the AI to have side specific skill settings? This overrides classname specific skills.
VCM_SKILL_CLASSNAMES = []; //Here you can assign certain unit classnames to specific skill levels. This will override the AI skill level above.

/*
EXAMPLE FOR VCM_SKILL_CLASSNAMES

VCM_SKILL_CLASSNAMES = [["Classname1",[aimingaccuracy,aimingshake,spotdistance,spottime,courage,commanding,aimingspeed,general,endurance,reloadspeed]],["Classname2",[aimingaccuracy,aimingshake,spotdistance,spottime,courage,commanding,aimingspeed,general,endurance,reloadspeed]]];
VCM_SKILL_CLASSNAMES = 	[
													["B_GEN_Soldier_F",[0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1]],
													["B_G_Soldier_AR_F",[0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1]]
												];

*/


//Set AI Skill levels
VCM_AIDIFSET =
{
	//Skip if Vcom Skillchange is disabled
	if (!VCM_SKILLCHANGE) exitWith {};
	{
		private _unit = _x;
		_unit setSkill 0.9;
		_unit allowFleeing 0;
		{
			_unit setSkill _x;
		} forEach VCM_AIDIFA;


		if (VCM_CLASSNAMESPECIFIC && {count VCM_SKILL_CLASSNAMES > 0}) then
		{
			{
				if (typeOf _unit isEqualTo (_x select 0)) exitWith
				{
					_ClassnameSet = true;
					_unit setSkill ["aimingAccuracy",((_x select 1) select 0)];_unit setSkill ["aimingShake",((_x select 1) select 1)];_unit setSkill ["spotDistance",((_x select 1) select 2)];_unit setSkill ["spotTime",((_x select 1) select 3)];/*_unit setSkill ["courage",((_x select 1) select 4)];_unit setSkill ["commanding",((_x select 1) select 5)];*/ _unit setSkill ["aimingSpeed",((_x select 1) select 6)];_unit setSkill ["general",((_x select 1) select 7)];_unit setSkill ["endurance",((_x select 1) select 8)];_unit setSkill ["reloadSpeed",((_x select 1) select 9)];
				};
			} foreach VCM_SKILL_CLASSNAMES;
		};

		if (VCM_SIDESPECIFICSKILL) then
		{
			_unit call VCM_AISIDESPEC;
		};

	} forEach (units _this);
};

diag_log "VCOM: Loaded Userconfig";


T8/CONFIG.hpp
/*
 =======================================================================================================================

	T8 Units Script
	
	Unit Spawn & Communication Script
	
	File:				CONFIG.hpp
	Author:				T-800a
	E-Mail:				t-800a@gmx.net
	
	init.sqf: 			[] execVM "T8_missionEXEC.sqf";
	
	description.ext:	#include <T8\CONFIG.hpp>
						class cfgFunctions
						{
							#include <\FUNCTIONS.hpp>
						};
						
	check  "T8_missionEXEC.sqf" to start spawning stuff...
		
	// Some information about variables saved to groups

	T8U_gvar_Comm 			-> [ bool, bool, bool ]		-> shares info, can be called, can react when attacked
	T8U_gvar_Origin			-> array					-> originTask: Marker, originTask: Order, ... 
	T8U_gvar_Settings		-> array or string			-> add. settings array defined in the units arrays
	T8U_gvar_Assigned		-> string					-> has Task Assigned: NO_TASK, CQC_SHOT, DC_ASSIST, ...
	T8U_gvar_Member			-> array					-> array of the origin units who are spawned
	T8U_gvar_Regrouped		-> bool						-> for GARRISON, PATROL_GARRISON: is set true when after group is released from task, now they can do their new task
	T8U_gvar_FiredEvent		-> array					-> Array filled by Fired Event
	T8U_gvar_Attacked		-> integer					-> time units of the group were last HIT ... if 'first time' force prone (0 targets) / suppressing fire (1++ targets)
	T8U_gvar_called			-> integer					-> time unit sent last call for help (general / set if other T8U group is called)
	T8U_gvar_DACcalled		-> integer					-> time Group last called DAC for HELP
	T8U_gvar_PARAcalled		-> integer					-> time Group last called for a Support (e.g. Para drop)
	T8U_gvar_ignoreGroup	-> bool						-> set by group handle if it shall ignore group (for quicker execution after 'first check')
	T8U_gvar_Introduced		-> bool						-> set true after editor groub is introduced to T8U
	
	T8U_ovar_lastSmoke
	T8U_gvar_behaviourSet
	T8U_gvar_lastBehaviour

	This work is licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
	To view a copy of this license, visit http://creativecommons.org/licenses/by-nc-sa/4.0/ or send a letter to 
	Creative Commons, 444 Castro Street, Suite 900, Mountain View, California, 94041, USA.

 =======================================================================================================================
*/

#define false			1
#define true			2

#define EAST			0
#define WEST			1
#define RESISTANCE		2
#define CIVILIAN		3


class cfgT8Units
{
	// ------------------------------------------------------------------------------
	// DEBUG Settings

	class debug
	{
		// general debug switch
		enable				= false;

		// show debug Hints
		allow_hints			= false;

		// create debug markers
		allow_marker		= false;

		// send debug messages to Killzone Kids console instead of RPT file
		// http://killzonekid.com/arma-console-extension-debug_console-dll-v3-0/
		allow_console		= false;	
	};


	// ------------------------------------------------------------------------------
	// MAIN configuration

	class main
	{
		// run the script on headless client and not on the Server
		// !! WARNING: this is untested !!
		use_HeadlessClient	= false;

		// option to register units with ZEUS
		// this may slow down spawning
		// will registers spawned units with "allCurators"
		allow_ZEUS			= true;

		// standard side for spawned units
		enemySide			= EAST;
		
		// selected modset for unit group compilation
		// see:		- class groupCompilations
		// 			- class groupRandomCompilations		
		modSet				= "vanilla";

		// RESISTANCE diplomacy 'switch'
		// for finding enemies and allies when calling for help - check with your mission settings
		//   0 : RESISTANCE neutral
		//   1 : RESISTANCE friendly to WEST
		//   2 : RESISTANCE friendly to EAST
		//   3 : RESISTANCE enemy of both; 
		diplomacy			= 1;
		
		// if group is in overwhelming superiority of XXX over their enemies, they wont call for help
		superiority			= 3;
		
		// the group leader shares info about known enemies to friendly units within XXX meter
		// knowsAbout > 1
		range_reveal		= 500;
		
		// group leader searches for help within XXX meter
		range_helpCall		= 1500;
		
		// a group can only call one other group each XXX seconds for help
		timeout_helpCall	= 60;
		
		// every XXX seconds a group can call in for a Support (e.g. para drop)
		// if the group has a squad leader or officer as their group leader
		timeout_support		= 180;
		
		// when the group is finished with their current task they return to their origin task after XXX seconds
		timeout_taskReturn	= 30;
		
		// units in zones are cached after XXX seconds when zone is deactivated
		timeout_caching		= 15;
		
		// zone radius + XXX meter is the distance in which units will patrol around a zone (waypoint positions affected)
		range_PatrolAround	= 50;
		
		// enable fatigue for ALL spawned units
		enable_fatigue		= false;
		
		// vehicles a group can use to travel greater distance (when they are called for help) 
		// if you want to allow vehicles from other addons, add them here
		reinforcementVehicles[] =
		{
			"APC_Tracked_01_base_F", "APC_Tracked_02_base_F", "Wheeled_APC_F", "Truck_01_base_F", "Truck_02_base_F", "MRAP_01_base_F",
			"MRAP_02_base_F", "MRAP_03_base_F", "C_Offroad_01_F", "I_G_Offroad_01_F"
		};

		// units that can lay down suppressive fire
		// if you want to allow units from other addons, add them here
		suppressingUnits[] = 
		{
			"B_soldier_AR_F", "B_G_soldier_AR_F", "O_soldier_AR_F", "O_soldierU_AR_F", "O_G_soldier_AR_F", "I_soldier_AR_F", "I_G_soldier_AR_F"
		};
		
		// ignored buildings for garrison and occupy
		ignoredBuildings[] = 
		{
			"Land_Garage_V1_ruins_F", "Land_i_Garage_V1_F", "Land_i_Garage_V1_dam_F", "Land_i_Garage_V2_F", "Land_i_Garage_V2_dam_F", "Land_i_Addon_03_V1_F", "Land_i_Addon_03mid_V1_F", "Land_i_Addon_04_V1_F"
		};

		// a group will go RED when in combat, then go GREEN after some time ( T8_UnitsEngageAtWillTime ), and then return to T8U_fnc_SpawnCombatMode
		enable_CBM			= false;
	};


	// ------------------------------------------------------------------------------
	// options to work with DAC by Silola
	
	class dac
	{
		// allow T8U to send a "call for help" to DAC
		enable				= false;
		
		// timeout for a group until it can sent the next "call for help"
		timeout				= 180;			
	};


	// ------------------------------------------------------------------------------
	// MAIN behavior and skills configuration
	
	class behaviorAndSkills
	{
/*
		# Skill Examples
		# militia ( untrained )
			aimingAccuracy	= 0.20;
			aimingShake		= 0.15;
			aimingSpeed		= 0.20;
			spotDistance	= 0.75;
			spotTime		= 0.70;
			courage			= 0.30;
			reloadSpeed		= 0.20;
			commanding		= 0.50;
			general			= 0.50;
	
		# regular forces
			aimingAccuracy	= 0.30;
			aimingShake		= 0.25;
			aimingSpeed		= 0.30;
			spotDistance	= 0.85;
			spotTime		= 0.75;
			courage			= 0.50;
			reloadSpeed		= 0.40;
			commanding		= 0.70;
			general			= 0.70;

		# special forces
			aimingAccuracy	= 0.45;
			aimingShake		= 0.40;
			aimingSpeed		= 0.50;
			spotDistance	= 0.95;
			spotTime		= 0.90;
			courage			= 0.70;
			reloadSpeed		= 0.60;
			commanding		= 0.90;
			general			= 0.90;

		# Behaivior Examples
		# aggressive
			behaivior[] =
			{
				"YELLOW",	// spawn Combat-Mode
				"RED",		// max. Combat-Mode when unit behaviour changes to COMBAT
				"WHITE",	// Combat-Mode after some time in max. Combat-Mode
				180			// time the group stays in max. Combat-Mode
			};

		# defensive
			behaivior[] = { "GREEN", "YELLOW", "GREEN", 90 };
	
		# medicore
			behaivior[] = { "GREEN", "RED", "GREEN", 120 };
*/	
		class west
		{
			class skills
			{
				class aimingAccuracy	{ value	= 0.45; };
				class aimingShake		{ value	= 0.40; };
				class aimingSpeed		{ value	= 0.50; };
				class spotDistance		{ value	= 0.95; };
				class spotTime			{ value	= 0.90; };
				class courage			{ value	= 0.70; };
				class reloadSpeed		{ value	= 0.60; };
				class commanding		{ value	= 0.90; };
				class general			{ value	= 0.90; };
			};
			
			behaivior[] = { "GREEN", "YELLOW", "GREEN", 90 };
		};
		
		class east
		{
			class skills
			{
				class aimingAccuracy	{ value	= 0.30; };
				class aimingShake		{ value	= 0.25; };
				class aimingSpeed		{ value	= 0.30; };
				class spotDistance		{ value	= 0.85; };
				class spotTime			{ value	= 0.75; };
				class courage			{ value	= 0.50; };
				class reloadSpeed		{ value	= 0.40; };
				class commanding		{ value	= 0.70; };
				class general			{ value	= 0.70; };
			};
			
			behaivior[] = { "YELLOW", "RED", "WHITE", 180 };
		};
		
		class guer
		{
			class skills
			{
				class aimingAccuracy	{ value	= 0.30; };
				class aimingShake		{ value	= 0.25; };
				class aimingSpeed		{ value	= 0.30; };
				class spotDistance		{ value	= 0.85; };
				class spotTime			{ value	= 0.75; };
				class courage			{ value	= 0.50; };
				class reloadSpeed		{ value	= 0.40; };
				class commanding		{ value	= 0.70; };
				class general			{ value	= 0.70; };
			};
			
			behaivior[] = { "GREEN", "RED", "GREEN", 120 };
		};
	};


	// ------------------------------------------------------------------------------
	// custom group settings

	class groupSettings
	{
		// some base classes
		class base_militia
		{
			class behaviorAndSkills
			{
				class skills
				{
					class aimingAccuracy	{ value	= 0.50; };
					class aimingShake		{ value	= 0.15; };
					class aimingSpeed		{ value	= 0.20; };
					class spotDistance		{ value	= 0.75; };
					class spotTime			{ value	= 0.70; };
					class courage			{ value	= 0.30; };
					class reloadSpeed		{ value	= 0.20; };
					class commanding		{ value	= 0.50; };
					class general			{ value	= 0.50; };
				};
				
				behaivior[] = { "GREEN", "RED", "GREEN", 120 };
			};
			
			teleport = false;
		};
		
		class base_regularForces
		{
			class behaviorAndSkills
			{
				class skills
				{
					class aimingAccuracy	{ value	= 0.30; };
					class aimingShake		{ value	= 0.25; };
					class aimingSpeed		{ value	= 0.30; };
					class spotDistance		{ value	= 0.85; };
					class spotTime			{ value	= 0.75; };
					class courage			{ value	= 0.50; };
					class reloadSpeed		{ value	= 0.40; };
					class commanding		{ value	= 0.70; };
					class general			{ value	= 0.70; };
				};
				
				behaivior[] = { "GREEN", "YELLOW", "GREEN", 90 };
			};
			
			teleport = false;
		};
		
		class base_specialForces
		{
			class behaviorAndSkills
			{
				class skills
				{
					class aimingAccuracy	{ value	= 0.45; };
					class aimingShake		{ value	= 0.40; };
					class aimingSpeed		{ value	= 0.50; };
					class spotDistance		{ value	= 0.95; };
					class spotTime			{ value	= 0.90; };
					class courage			{ value	= 0.70; };
					class reloadSpeed		{ value	= 0.60; };
					class commanding		{ value	= 0.90; };
					class general			{ value	= 0.90; };
				};
				
				behaivior[] = { "YELLOW", "RED", "WHITE", 180 };
			};
			
			teleport = false;
		};
		
		// 
		class teleport : base_militia
		{
			teleport = true;
		};
	};


	// ------------------------------------------------------------------------------
	// configuration for fixed unit containers	

	class groupCompilations
	{
		class vanilla
		{
			class west
			{
				fireteam[]	= {	"B_soldier_SL_F", "B_medic_F", "O_soldier_F", "B_soldier_AR_F" };
				squad[]		= {	"B_soldier_SL_F", "B_medic_F", "O_soldier_F", "B_soldier_AR_F", "B_soldier_TL_F", "B_soldier_F", "B_soldier_F", "B_soldier_LAT_F" };
				platoon[]	= {	"B_officer_F", "B_medic_F", "O_soldier_F", "B_soldier_AR_F", "B_soldier_TL_F", "B_soldier_F", "B_soldier_F", "B_soldier_LAT_F",
								"B_soldier_SL_F", "B_medic_F", "O_soldier_F", "B_soldier_AR_F", "B_soldier_TL_F", "B_soldier_F", "B_soldier_F", "B_soldier_LAT_F" };
			};
			
			class east
			{
				fireteam[]	= { "O_soldier_SL_F", "O_medic_F", "O_soldier_F", "O_soldier_AR_F" };
				squad[]		= { "O_soldier_SL_F", "O_medic_F", "O_soldier_F", "O_soldier_AR_F", "O_soldier_TL_F", "O_soldier_F", "O_soldier_F", "O_soldier_LAT_F" };
				platoon[]	= {	"O_officer_F", "O_medic_F", "O_soldier_F", "O_soldier_AR_F", "O_soldier_TL_F", "O_soldier_F", "O_soldier_F", "O_soldier_LAT_F",
								"O_soldier_SL_F", "O_medic_F", "O_soldier_F", "O_soldier_AR_F", "O_soldier_TL_F", "O_soldier_F", "O_soldier_F", "O_soldier_LAT_F" };
			};
			
			class guer
			{
				fireteam[]	= { "I_soldier_SL_F", "I_medic_F", "I_soldier_F", "I_soldier_AR_F" };
				squad[]		= { "I_soldier_SL_F", "I_medic_F", "I_soldier_F", "I_soldier_AR_F", "I_soldier_TL_F", "I_soldier_F", "I_soldier_F", "I_soldier_LAT_F" };
				platoon[]	= {	"I_officer_F", "I_medic_F", "I_soldier_F", "I_soldier_AR_F", "I_soldier_TL_F", "I_soldier_F", "I_soldier_F", "I_soldier_LAT_F",
								"I_soldier_SL_F", "I_medic_F", "I_soldier_F", "I_soldier_AR_F", "I_soldier_TL_F", "I_soldier_F", "I_soldier_F", "I_soldier_LAT_F" };
			};
			
			class civ
			{
				fireteam[]	= { "C_man_1_1_F", "C_man_1_2_F", "C_man_polo_1_F", "C_man_polo_2_F" };
				squad[]		= { "C_man_p_fugitive_F", "C_man_polo_1_F_asia", "C_man_1_1_F", "C_man_1_2_F", "C_man_1_3_F", "C_man_polo_1_F", "C_man_polo_2_F", "C_man_polo_3_F" };
				platoon[]	= { "C_man_p_fugitive_F", "C_man_polo_1_F_asia", "C_man_1_1_F", "C_man_1_2_F", "C_man_1_3_F", "C_man_polo_1_F", "C_man_polo_2_F", "C_man_polo_3_F",
								"C_man_p_fugitive_F", "C_man_polo_1_F_asia", "C_man_1_1_F", "C_man_1_2_F", "C_man_1_3_F", "C_man_polo_1_F", "C_man_polo_2_F", "C_man_polo_3_F" };
			};
		};
	};


	// ------------------------------------------------------------------------------
	// configuration for randomized unit containers
	//
	// if you want a randomized selection of units you can use the following function in your units array:
	//    [ "east", "infantry", 5, 3 ] call T8U_fnc_createUnitContainer
	// units selected by this function are defined below
	// if you add your custom selections (as example for different mods), please share them with me!

	class groupRandomCompilations
	{
		class vanilla
		{
			class west
			{
				infantry[]		= { "B_Soldier_F","B_Soldier_02_f","B_Soldier_03_f","B_Soldier_04_f","B_Soldier_05_f","B_Soldier_lite_F","B_Soldier_GL_F","B_soldier_AR_F","B_Soldier_SL_F","B_Soldier_TL_F","B_soldier_M_F","B_soldier_LAT_F","B_medic_F","B_soldier_repair_F","B_soldier_exp_F","B_Soldier_A_F","B_soldier_AT_F","B_soldier_AA_F","B_engineer_F","B_officer_F","B_soldier_PG_F","B_recon_F","B_recon_LAT_F","B_recon_exp_F","B_recon_medic_F","B_recon_TL_F","B_recon_M_F","B_recon_JTAC_F","B_spotter_F","B_sniper_F","B_support_MG_F","B_support_GMG_F","B_support_Mort_F","B_support_AMG_F","B_support_AMort_F","B_ghillie_lsh_F","B_ghillie_sard_F","B_ghillie_ard_F","B_Sharpshooter_F","B_Recon_Sharpshooter_F","B_HeavyGunner_F","B_G_Sharpshooter_F" };
				vehicles[]		= { "B_APC_Tracked_01_AA_F","B_MBT_01_cannon_F","B_MBT_01_cannon_F","B_MRAP_01_gmg_F","B_MRAP_01_hmg_F","B_G_Offroad_01_armed_F","B_APC_Wheeled_01_cannon_F","B_MBT_01_TUSK_F" };
				artillery[]		= { "B_MBT_01_mlrs_F","B_MBT_01_arty_F" };
				helicopter[]	= { "B_Heli_Light_01_armed_F","B_Heli_Attack_01_F" };
				air[]			= { "B_Plane_CAS_01_F" };
			};

			class east
			{
				infantry[]		= { "O_G_Soldier_F","O_G_Soldier_lite_F","O_G_Soldier_SL_F","O_G_Soldier_TL_F","O_G_Soldier_AR_F","O_G_medic_F","O_G_engineer_F","O_G_Soldier_exp_F","O_G_Soldier_GL_F","O_G_Soldier_M_F","O_G_Soldier_LAT_F","O_G_Soldier_A_F","O_G_officer_F","O_Soldier_F","O_officer_F","O_Soldier_lite_F","O_Soldier_GL_F","O_Soldier_AR_F","O_Soldier_SL_F", "O_Soldier_TL_F","O_soldier_M_F","O_Soldier_LAT_F","O_medic_F","O_soldier_repair_F","O_soldier_exp_F","O_Soldier_A_F","O_Soldier_AT_F","O_Soldier_AA_F","O_engineer_F","O_Sharpshooter_F","O_HeavyGunner_F","O_G_Sharpshooter_F","O_recon_F","O_recon_M_F","O_recon_LAT_F","O_recon_medic_F","O_recon_exp_F","O_recon_JTAC_F", "O_recon_TL_F","O_Pathfinder_F","O_sniper_F","O_ghillie_lsh_F","O_ghillie_sard_F","O_ghillie_ard_F" };
				vehicles[]		= { "O_APC_Tracked_02_cannon_F","O_APC_Tracked_02_AA_F","O_MBT_02_cannon_F","O_MRAP_02_hmg_F","O_MRAP_02_gmg_F","O_G_Offroad_01_armed_F","O_APC_Wheeled_02_rcws_F" };
				artillery[]		= { "O_MBT_02_arty_F" };
				helicopter[]	= { "O_Heli_Light_02_F","O_Heli_Light_02_v2_F","O_Heli_Attack_02_F","O_Heli_Attack_02_black_F" };
				air[]			= { "O_Plane_CAS_02_F" };
			};

			class guer
			{
				infantry[]		= { "I_G_Soldier_F","I_G_Soldier_lite_F","I_G_Soldier_SL_F","I_G_Soldier_TL_F","I_G_Soldier_AR_F","  	I_G_medic_F","I_G_engineer_F","I_G_Soldier_exp_F","I_G_Soldier_GL_F","I_G_Soldier_M_F","I_G_Soldier_LAT_F","I_G_Soldier_A_F","I_G_officer_F","I_Soldier_02_F","I_Soldier_03_F","I_Soldier_04_F","I_soldier_F","  	I_Soldier_lite_F","I_Soldier_A_F","I_Soldier_GL_F","I_Soldier_AR_F","I_Soldier_SL_F","I_Soldier_TL_F","I_Soldier_M_F","I_Soldier_LAT_F","I_Soldier_AT_F","I_Soldier_AA_F","I_medic_F","I_Soldier_repair_F","I_Soldier_exp_F","I_engineer_F","I_officer_F","I_Spotter_F","I_Sniper_F","I_Soldier_AAR_F","I_Soldier_AAT_F","I_Soldier_AAA_F","I_support_MG_F","I_support_GMG_F","I_support_Mort_F","I_support_AMG_F","I_support_AMort_F"," I_ghillie_lsh_F","I_ghillie_sard_F","I_ghillie_ard_F","I_G_Sharpshooter_F" };
				vehicles[]		= { "I_APC_tracked_03_cannon_F","I_MBT_03_cannon_F","I_G_Offroad_01_armed_F","I_MRAP_03_hmg_F","I_MRAP_03_gmg_F","I_APC_Wheeled_03_cannon_F" };
				artillery[]		= { "I_MBT_03_cannon_F" };
				helicopter[]	= { "I_Heli_light_03_F" };
				air[]			= { "I_Plane_Fighter_03_CAS_F","I_Plane_Fighter_03_AA_F" };
			};
			
			class civ
			{
				infantry[]		= {};
				vehicles[]		= {};
				artillery[]		= {};
				helicopter[]	= {};
				air[]			= {};
			};
		};
	};
};





///// END OF CONFIG FILE ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////